src/Microsoft.Net.Http.Headers/DateTimeFormatter.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using static System.Text.Encoding;
+
+namespace Microsoft.Net.Http.Headers
+{
+    internal static class DateTimeFormatter
```

Comment:
You could also consider vectorizing this where the lengths are known, using `(int *)` or `(long *)`. I think that what's there is probably fine, and this is important it would come up again.